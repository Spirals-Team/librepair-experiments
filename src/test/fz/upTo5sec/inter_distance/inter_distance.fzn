array [1..2] of int: X_INTRODUCED_8 = [1,-1];
array [1..2] of int: X_INTRODUCED_11 = [-1,1];
var 0..32: X_INTRODUCED_0;
var 0..32: X_INTRODUCED_1;
var 0..32: X_INTRODUCED_2;
var 0..32: X_INTRODUCED_3;
var 0..32: X_INTRODUCED_4;
var 0..32: X_INTRODUCED_5;
var 0..32: X_INTRODUCED_6;
var 0..32: X_INTRODUCED_7;
var 2..10: p:: output_var;
var -32..32: X_INTRODUCED_9 ::var_is_introduced :: is_defined_var;
var 0..32: X_INTRODUCED_10 ::var_is_introduced ;
var -32..32: X_INTRODUCED_12 ::var_is_introduced :: is_defined_var;
var 0..32: X_INTRODUCED_13 ::var_is_introduced ;
var -32..32: X_INTRODUCED_14 ::var_is_introduced :: is_defined_var;
var 0..32: X_INTRODUCED_15 ::var_is_introduced ;
var -32..32: X_INTRODUCED_16 ::var_is_introduced :: is_defined_var;
var 0..32: X_INTRODUCED_17 ::var_is_introduced ;
var -32..32: X_INTRODUCED_18 ::var_is_introduced :: is_defined_var;
var 0..32: X_INTRODUCED_19 ::var_is_introduced ;
var -32..32: X_INTRODUCED_20 ::var_is_introduced :: is_defined_var;
var 0..32: X_INTRODUCED_21 ::var_is_introduced ;
var -32..32: X_INTRODUCED_22 ::var_is_introduced :: is_defined_var;
var 0..32: X_INTRODUCED_23 ::var_is_introduced ;
var -32..32: X_INTRODUCED_24 ::var_is_introduced :: is_defined_var;
var 0..32: X_INTRODUCED_25 ::var_is_introduced ;
var -32..32: X_INTRODUCED_26 ::var_is_introduced :: is_defined_var;
var 0..32: X_INTRODUCED_27 ::var_is_introduced ;
var -32..32: X_INTRODUCED_28 ::var_is_introduced :: is_defined_var;
var 0..32: X_INTRODUCED_29 ::var_is_introduced ;
var -32..32: X_INTRODUCED_30 ::var_is_introduced :: is_defined_var;
var 0..32: X_INTRODUCED_31 ::var_is_introduced ;
var -32..32: X_INTRODUCED_32 ::var_is_introduced :: is_defined_var;
var 0..32: X_INTRODUCED_33 ::var_is_introduced ;
var -32..32: X_INTRODUCED_34 ::var_is_introduced :: is_defined_var;
var 0..32: X_INTRODUCED_35 ::var_is_introduced ;
var -32..32: X_INTRODUCED_36 ::var_is_introduced :: is_defined_var;
var 0..32: X_INTRODUCED_37 ::var_is_introduced ;
var -32..32: X_INTRODUCED_38 ::var_is_introduced :: is_defined_var;
var 0..32: X_INTRODUCED_39 ::var_is_introduced ;
var -32..32: X_INTRODUCED_40 ::var_is_introduced :: is_defined_var;
var 0..32: X_INTRODUCED_41 ::var_is_introduced ;
var -32..32: X_INTRODUCED_42 ::var_is_introduced :: is_defined_var;
var 0..32: X_INTRODUCED_43 ::var_is_introduced ;
var -32..32: X_INTRODUCED_44 ::var_is_introduced :: is_defined_var;
var 0..32: X_INTRODUCED_45 ::var_is_introduced ;
var -32..32: X_INTRODUCED_46 ::var_is_introduced :: is_defined_var;
var 0..32: X_INTRODUCED_47 ::var_is_introduced ;
var -32..32: X_INTRODUCED_48 ::var_is_introduced :: is_defined_var;
var 0..32: X_INTRODUCED_49 ::var_is_introduced ;
var -32..32: X_INTRODUCED_50 ::var_is_introduced :: is_defined_var;
var 0..32: X_INTRODUCED_51 ::var_is_introduced ;
var -32..32: X_INTRODUCED_52 ::var_is_introduced :: is_defined_var;
var 0..32: X_INTRODUCED_53 ::var_is_introduced ;
var -32..32: X_INTRODUCED_54 ::var_is_introduced :: is_defined_var;
var 0..32: X_INTRODUCED_55 ::var_is_introduced ;
var -32..32: X_INTRODUCED_56 ::var_is_introduced :: is_defined_var;
var 0..32: X_INTRODUCED_57 ::var_is_introduced ;
var -32..32: X_INTRODUCED_58 ::var_is_introduced :: is_defined_var;
var 0..32: X_INTRODUCED_59 ::var_is_introduced ;
var -32..32: X_INTRODUCED_60 ::var_is_introduced :: is_defined_var;
var 0..32: X_INTRODUCED_61 ::var_is_introduced ;
var -32..32: X_INTRODUCED_62 ::var_is_introduced :: is_defined_var;
var 0..32: X_INTRODUCED_63 ::var_is_introduced ;
var -32..32: X_INTRODUCED_64 ::var_is_introduced :: is_defined_var;
var 0..32: X_INTRODUCED_65 ::var_is_introduced ;
array [1..8] of var int: x:: output_array([1..8]) = [X_INTRODUCED_0,X_INTRODUCED_1,X_INTRODUCED_2,X_INTRODUCED_3,X_INTRODUCED_4,X_INTRODUCED_5,X_INTRODUCED_6,X_INTRODUCED_7];
constraint int_abs(X_INTRODUCED_9,X_INTRODUCED_10);
constraint int_lin_le(X_INTRODUCED_11,[X_INTRODUCED_10,p],0);
constraint int_abs(X_INTRODUCED_12,X_INTRODUCED_13);
constraint int_lin_le(X_INTRODUCED_11,[X_INTRODUCED_13,p],0);
constraint int_abs(X_INTRODUCED_14,X_INTRODUCED_15);
constraint int_lin_le(X_INTRODUCED_11,[X_INTRODUCED_15,p],0);
constraint int_abs(X_INTRODUCED_16,X_INTRODUCED_17);
constraint int_lin_le(X_INTRODUCED_11,[X_INTRODUCED_17,p],0);
constraint int_abs(X_INTRODUCED_18,X_INTRODUCED_19);
constraint int_lin_le(X_INTRODUCED_11,[X_INTRODUCED_19,p],0);
constraint int_abs(X_INTRODUCED_20,X_INTRODUCED_21);
constraint int_lin_le(X_INTRODUCED_11,[X_INTRODUCED_21,p],0);
constraint int_abs(X_INTRODUCED_22,X_INTRODUCED_23);
constraint int_lin_le(X_INTRODUCED_11,[X_INTRODUCED_23,p],0);
constraint int_abs(X_INTRODUCED_24,X_INTRODUCED_25);
constraint int_lin_le(X_INTRODUCED_11,[X_INTRODUCED_25,p],0);
constraint int_abs(X_INTRODUCED_26,X_INTRODUCED_27);
constraint int_lin_le(X_INTRODUCED_11,[X_INTRODUCED_27,p],0);
constraint int_abs(X_INTRODUCED_28,X_INTRODUCED_29);
constraint int_lin_le(X_INTRODUCED_11,[X_INTRODUCED_29,p],0);
constraint int_abs(X_INTRODUCED_30,X_INTRODUCED_31);
constraint int_lin_le(X_INTRODUCED_11,[X_INTRODUCED_31,p],0);
constraint int_abs(X_INTRODUCED_32,X_INTRODUCED_33);
constraint int_lin_le(X_INTRODUCED_11,[X_INTRODUCED_33,p],0);
constraint int_abs(X_INTRODUCED_34,X_INTRODUCED_35);
constraint int_lin_le(X_INTRODUCED_11,[X_INTRODUCED_35,p],0);
constraint int_abs(X_INTRODUCED_36,X_INTRODUCED_37);
constraint int_lin_le(X_INTRODUCED_11,[X_INTRODUCED_37,p],0);
constraint int_abs(X_INTRODUCED_38,X_INTRODUCED_39);
constraint int_lin_le(X_INTRODUCED_11,[X_INTRODUCED_39,p],0);
constraint int_abs(X_INTRODUCED_40,X_INTRODUCED_41);
constraint int_lin_le(X_INTRODUCED_11,[X_INTRODUCED_41,p],0);
constraint int_abs(X_INTRODUCED_42,X_INTRODUCED_43);
constraint int_lin_le(X_INTRODUCED_11,[X_INTRODUCED_43,p],0);
constraint int_abs(X_INTRODUCED_44,X_INTRODUCED_45);
constraint int_lin_le(X_INTRODUCED_11,[X_INTRODUCED_45,p],0);
constraint int_abs(X_INTRODUCED_46,X_INTRODUCED_47);
constraint int_lin_le(X_INTRODUCED_11,[X_INTRODUCED_47,p],0);
constraint int_abs(X_INTRODUCED_48,X_INTRODUCED_49);
constraint int_lin_le(X_INTRODUCED_11,[X_INTRODUCED_49,p],0);
constraint int_abs(X_INTRODUCED_50,X_INTRODUCED_51);
constraint int_lin_le(X_INTRODUCED_11,[X_INTRODUCED_51,p],0);
constraint int_abs(X_INTRODUCED_52,X_INTRODUCED_53);
constraint int_lin_le(X_INTRODUCED_11,[X_INTRODUCED_53,p],0);
constraint int_abs(X_INTRODUCED_54,X_INTRODUCED_55);
constraint int_lin_le(X_INTRODUCED_11,[X_INTRODUCED_55,p],0);
constraint int_abs(X_INTRODUCED_56,X_INTRODUCED_57);
constraint int_lin_le(X_INTRODUCED_11,[X_INTRODUCED_57,p],0);
constraint int_abs(X_INTRODUCED_58,X_INTRODUCED_59);
constraint int_lin_le(X_INTRODUCED_11,[X_INTRODUCED_59,p],0);
constraint int_abs(X_INTRODUCED_60,X_INTRODUCED_61);
constraint int_lin_le(X_INTRODUCED_11,[X_INTRODUCED_61,p],0);
constraint int_abs(X_INTRODUCED_62,X_INTRODUCED_63);
constraint int_lin_le(X_INTRODUCED_11,[X_INTRODUCED_63,p],0);
constraint int_abs(X_INTRODUCED_64,X_INTRODUCED_65);
constraint int_lin_le(X_INTRODUCED_11,[X_INTRODUCED_65,p],0);
constraint int_lin_le(X_INTRODUCED_8,[X_INTRODUCED_0,X_INTRODUCED_1],0);
constraint int_lin_le(X_INTRODUCED_8,[X_INTRODUCED_1,X_INTRODUCED_2],0);
constraint int_lin_le(X_INTRODUCED_8,[X_INTRODUCED_2,X_INTRODUCED_3],0);
constraint int_lin_le(X_INTRODUCED_8,[X_INTRODUCED_3,X_INTRODUCED_4],0);
constraint int_lin_le(X_INTRODUCED_8,[X_INTRODUCED_4,X_INTRODUCED_5],0);
constraint int_lin_le(X_INTRODUCED_8,[X_INTRODUCED_5,X_INTRODUCED_6],0);
constraint int_lin_le(X_INTRODUCED_8,[X_INTRODUCED_6,X_INTRODUCED_7],0);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_0,X_INTRODUCED_1,X_INTRODUCED_9],0):: defines_var(X_INTRODUCED_9);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_0,X_INTRODUCED_2,X_INTRODUCED_12],0):: defines_var(X_INTRODUCED_12);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_0,X_INTRODUCED_3,X_INTRODUCED_14],0):: defines_var(X_INTRODUCED_14);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_0,X_INTRODUCED_4,X_INTRODUCED_16],0):: defines_var(X_INTRODUCED_16);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_0,X_INTRODUCED_5,X_INTRODUCED_18],0):: defines_var(X_INTRODUCED_18);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_0,X_INTRODUCED_6,X_INTRODUCED_20],0):: defines_var(X_INTRODUCED_20);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_0,X_INTRODUCED_7,X_INTRODUCED_22],0):: defines_var(X_INTRODUCED_22);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_1,X_INTRODUCED_2,X_INTRODUCED_24],0):: defines_var(X_INTRODUCED_24);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_1,X_INTRODUCED_3,X_INTRODUCED_26],0):: defines_var(X_INTRODUCED_26);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_1,X_INTRODUCED_4,X_INTRODUCED_28],0):: defines_var(X_INTRODUCED_28);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_1,X_INTRODUCED_5,X_INTRODUCED_30],0):: defines_var(X_INTRODUCED_30);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_1,X_INTRODUCED_6,X_INTRODUCED_32],0):: defines_var(X_INTRODUCED_32);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_1,X_INTRODUCED_7,X_INTRODUCED_34],0):: defines_var(X_INTRODUCED_34);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_2,X_INTRODUCED_3,X_INTRODUCED_36],0):: defines_var(X_INTRODUCED_36);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_2,X_INTRODUCED_4,X_INTRODUCED_38],0):: defines_var(X_INTRODUCED_38);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_2,X_INTRODUCED_5,X_INTRODUCED_40],0):: defines_var(X_INTRODUCED_40);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_2,X_INTRODUCED_6,X_INTRODUCED_42],0):: defines_var(X_INTRODUCED_42);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_2,X_INTRODUCED_7,X_INTRODUCED_44],0):: defines_var(X_INTRODUCED_44);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_3,X_INTRODUCED_4,X_INTRODUCED_46],0):: defines_var(X_INTRODUCED_46);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_3,X_INTRODUCED_5,X_INTRODUCED_48],0):: defines_var(X_INTRODUCED_48);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_3,X_INTRODUCED_6,X_INTRODUCED_50],0):: defines_var(X_INTRODUCED_50);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_3,X_INTRODUCED_7,X_INTRODUCED_52],0):: defines_var(X_INTRODUCED_52);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_4,X_INTRODUCED_5,X_INTRODUCED_54],0):: defines_var(X_INTRODUCED_54);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_4,X_INTRODUCED_6,X_INTRODUCED_56],0):: defines_var(X_INTRODUCED_56);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_4,X_INTRODUCED_7,X_INTRODUCED_58],0):: defines_var(X_INTRODUCED_58);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_5,X_INTRODUCED_6,X_INTRODUCED_60],0):: defines_var(X_INTRODUCED_60);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_5,X_INTRODUCED_7,X_INTRODUCED_62],0):: defines_var(X_INTRODUCED_62);
constraint int_lin_eq([1,-1,-1],[X_INTRODUCED_6,X_INTRODUCED_7,X_INTRODUCED_64],0):: defines_var(X_INTRODUCED_64);
solve  maximize p;
