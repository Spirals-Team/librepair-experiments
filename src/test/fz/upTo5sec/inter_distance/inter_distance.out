p = 4;
x = array1d(1..8, [0, 4, 8, 12, 16, 20, 24, 28]);
----------
==========